--- a/browser/branding/unofficial/configure.sh	2019-05-17 02:33:15.000000000 +0200
+++ b/browser/branding/unofficial/configure.sh	2019-05-22 19:50:41.253784229 +0200
@@ -2,4 +2,5 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-MOZ_APP_DISPLAYNAME=Nightly
+MOZ_APP_DISPLAYNAME=plasmafox
+MOZ_APP_PROFILE=plasmafox
--- a/browser/base/content/overrides/app-license.html	2019-05-17 02:33:14.000000000 +0200
+++ b/browser/base/content/overrides/app-license.html	2019-05-24 15:48:44.718916525 +0200
@@ -2,5 +2,5 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/.  -->
     <p><b>Binaries</b> of this product have been made available to you by the
-    <a href="http://www.mozilla.org/">Mozilla Project</a> under the Mozilla
+    <a href="http://github.com/torvic9/plasmafox">Plasmafox Project</a> under the Mozilla
     Public License 2.0 (MPL). <a href="about:rights">Know your rights</a>.</p>

--- a/browser/base/content/aboutDialog.xul	2019-05-17 02:33:13.000000000 +0200
+++ b/browser/base/content/aboutDialog.xul	2019-05-24 17:23:38.504728883 +0200
@@ -133,12 +133,12 @@
             </description>
           </vbox>
           <description class="text-blurb" id="communityDesc" data-l10n-id="community-2">
-            <label is="text-link" href="http://www.mozilla.org/" data-l10n-name="community-mozillaLink"></label>
+            <label is="text-link" href="https://github.com/torvic9/plasmafox" data-l10n-name="community-mozillaLink"></label>
             <label is="text-link" useoriginprincipal="true" href="about:credits" data-l10n-name="community-creditsLink"></label>
           </description>
           <description class="text-blurb" id="contributeDesc" data-l10n-id="helpus">
-            <label is="text-link" href="https://donate.mozilla.org/?utm_source=firefox&#38;utm_medium=referral&#38;utm_campaign=firefox_about&#38;utm_content=firefox_about" data-l10n-name="helpus-donateLink"></label>
-            <label is="text-link" href="http://www.mozilla.org/contribute/" data-l10n-name="helpus-getInvolvedLink"></label>
+            <label is="text-link" href="https://github.com/torvic9/plasmafox" data-l10n-name="helpus-donateLink"></label>
+            <label is="text-link" href="https://github.com/torvic9/plasmafox" data-l10n-name="helpus-getInvolvedLink"></label>
           </description>
         </vbox>
       </vbox>
--- a/browser/locales/en-US/browser/aboutDialog.ftl	2019-05-17 02:33:18.000000000 +0200
+++ b/browser/locales/en-US/browser/aboutDialog.ftl	2019-05-24 17:18:41.000000000 +0200
@@ -39,7 +39,7 @@
 
 community-exp = <label data-l10n-name="community-exp-mozillaLink">{ -vendor-short-name }</label> is a <label data-l10n-name="community-exp-creditsLink">global community</label> working together to keep the Web open, public and accessible to all.
 
-community-2 = { -brand-short-name } is designed by <label data-l10n-name="community-mozillaLink">{ -vendor-short-name }</label>, a <label data-l10n-name="community-creditsLink">global community</label> working together to keep the Web open, public and accessible to all.
+community-2 = { -brand-short-name } is designed by <label data-l10n-name="community-mozillaLink">{ -vendor-short-name }</label>, to be used with Manjaro KDE.
 
 helpus = Want to help? <label data-l10n-name="helpus-donateLink">Make a donation</label> or <label data-l10n-name="helpus-getInvolvedLink">get involved!</label>
 
--- a/browser/installer/package-manifest.in	2019-05-17 02:33:18.000000000 +0200
+++ b/browser/installer/package-manifest.in	2019-05-24 17:57:01.733426808 +0200
@@ -242,10 +242,13 @@
 @RESPATH@/chrome/recording/*
 #ifdef MOZ_GTK
 @RESPATH@/browser/chrome/icons/default/default16.png
+@RESPATH@/browser/chrome/icons/default/default22.png
+@RESPATH@/browser/chrome/icons/default/default24.png
 @RESPATH@/browser/chrome/icons/default/default32.png
 @RESPATH@/browser/chrome/icons/default/default48.png
 @RESPATH@/browser/chrome/icons/default/default64.png
 @RESPATH@/browser/chrome/icons/default/default128.png
+@RESPATH@/browser/chrome/icons/default/default256.png
 #endif
 @RESPATH@/browser/features/*
 
--- a/browser/branding/branding-common.mozbuild	2019-05-17 02:33:15.000000000 +0200
+++ b/browser/branding/branding-common.mozbuild	2019-05-24 17:55:58.879724433 +0200
@@ -27,6 +27,9 @@
         FINAL_TARGET_FILES.chrome.icons.default += [
             'default128.png',
             'default16.png',
+            'default22.png',
+            'default24.png',
+            'default256.png',
             'default32.png',
             'default48.png',
             'default64.png',
--- a/browser/branding/unofficial/locales/en-US/brand.dtd	2019-05-17 02:33:15.000000000 +0200
+++ b/browser/branding/unofficial/locales/en-US/brand.dtd	2019-05-22 23:17:10.176572173 +0200
@@ -2,12 +2,12 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
-<!ENTITY  brandShorterName      "Nightly">
-<!ENTITY  brandShortName        "Nightly">
-<!ENTITY  brandFullName         "Nightly">
+<!ENTITY  brandShorterName      "Plasmafox">
+<!ENTITY  brandShortName        "Plasmafox">
+<!ENTITY  brandFullName         "Plasmafox">
 <!-- LOCALIZATION NOTE (brandProductName):
    This brand name can be used in messages where the product name needs to
    remain unchanged across different versions (Nightly, Beta, etc.). -->
-<!ENTITY  brandProductName      "Firefox">
-<!ENTITY  vendorShortName       "Mozilla">
-<!ENTITY  trademarkInfo.part1   " ">
+<!ENTITY  brandProductName      "Plasmafox">
+<!ENTITY  vendorShortName       "Plasmafox">
+<!ENTITY  trademarkInfo.part1   "Plasmafox is not associated with Mozilla">
--- a/browser/branding/unofficial/locales/en-US/brand.ftl	2019-05-17 02:33:14.000000000 +0200
+++ b/browser/branding/unofficial/locales/en-US/brand.ftl	2019-05-22 23:17:39.947484171 +0200
@@ -12,11 +12,11 @@
 ##
 ## Reference: https://www.mozilla.org/styleguide/communications/translation/
 
--brand-shorter-name = Nightly
--brand-short-name = Nightly
--brand-full-name = Nightly
+-brand-shorter-name = Plasmafox
+-brand-short-name = Plasmafox
+-brand-full-name = Plasmafox
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
-trademarkInfo = { " " }
+-brand-product-name = Plasmafox
+-vendor-short-name = Plasmafox
+trademarkInfo = { "Plasmafox is not associated with Mozilla" }
--- a/browser/branding/unofficial/locales/en-US/brand.properties	2019-05-17 02:33:15.000000000 +0200
+++ b/browser/branding/unofficial/locales/en-US/brand.properties	2019-05-22 23:16:16.441536433 +0200
@@ -2,13 +2,13 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-brandShorterName=Nightly
-brandShortName=Nightly
-brandFullName=Nightly
+brandShorterName=Plasmafox
+brandShortName=Plasmafox
+brandFullName=Plasmafox
 # LOCALIZATION NOTE(brandProductName):
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
-brandProductName=Firefox
-vendorShortName=Mozilla
+brandProductName=Plasmafox
+vendorShortName=Plasmafox
 
 syncBrandShortName=Sync
--- a/browser/themes/shared/privatebrowsing/aboutPrivateBrowsing.css	2019-05-17 02:33:19.000000000 +0200
+++ b/browser/themes/shared/privatebrowsing/aboutPrivateBrowsing.css	2019-05-24 21:57:51.438173329 +0200
@@ -124,7 +124,7 @@
 }
 
 .wordmark {
-  background: url("resource://activity-stream/data/content/assets/firefox-wordmark.svg") no-repeat center center;
+  background: url("resource://activity-stream/data/content/assets/plasmafox-wordmark.svg") no-repeat center center;
   background-size: 175px;
   -moz-context-properties: fill;
   display: inline-block;

--- a/browser/branding/unofficial/pref/firefox-branding.js	2019-05-29 17:38:45.000000000 +0200
+++ b/browser/branding/unofficial/pref/firefox-branding.js	2019-06-07 20:48:04.084846667 +0200
@@ -2,8 +2,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-pref("startup.homepage_override_url", "");
-pref("startup.homepage_welcome_url", "");
+pref("startup.homepage_override_url", "https://github.com/torvic9/plasmafox");
+pref("startup.homepage_welcome_url", "https://github.com/torvic9/plasmafox");
 pref("startup.homepage_welcome_url.additional", "");
 // The time interval between checks for a new version (in seconds)
 pref("app.update.interval", 86400); // 24 hours
@@ -11,10 +11,10 @@
 pref("app.update.promptWaitTime", 86400);
 // URL user can browse to manually if for some reason all update installation
 // attempts fail.
-pref("app.update.url.manual", "https://nightly.mozilla.org");
+pref("app.update.url.manual", "https://github.com/torvic9/plasmafox");
 // A default value for the "More information about this update" link
 // supplied in the "An update is available" page of the update wizard.
-pref("app.update.url.details", "https://nightly.mozilla.org");
+pref("app.update.url.details", "https://github.com/torvic9/plasmafox");
 
 // The number of days a binary is permitted to be old
 // without checking for an update.  This assumes that
--- a/browser/app/profile/firefox.js	2019-05-29 17:38:45.000000000 +0200
+++ b/browser/app/profile/firefox.js	2019-06-07 20:54:16.736578935 +0200
@@ -391,6 +391,10 @@
 // Mirrors whether the search-container widget is in the navigation toolbar.
 pref("browser.search.widget.inNavBar", false);
 
+// Search defaults
+pref("browser.search.defaultenginename.US", "data:text/plain,browser.search.defaultenginename.US=DuckDuckGo");
+pref("browser.search.defaultenginename", "DuckDuckGo");
+pref("browser.search.suggest.enabled", false);
 #ifndef RELEASE_OR_BETA
 pref("browser.search.reset.enabled", true);
 #endif
@@ -1777,7 +1781,7 @@
 pref("browser.suppress_first_window_animation", true);
 
 // Preference that allows individual users to disable Screenshots.
-pref("extensions.screenshots.disabled", false);
+//pref("extensions.screenshots.disabled", false);
 // Preference that allows individual users to leave Screenshots enabled, but
 // disable uploading to the server.
 pref("extensions.screenshots.upload-disabled", false);
--- a/browser/locales/en-US/chrome/browser-region/region.properties	2019-05-29 17:38:45.000000000 +0200
+++ b/browser/locales/en-US/chrome/browser-region/region.properties	2019-06-07 20:52:50.251726354 +0200
@@ -22,3 +22,7 @@
 # The default set of protocol handlers for ircs:
 gecko.handlerService.schemes.ircs.0.name=Mibbit
 gecko.handlerService.schemes.ircs.0.uriTemplate=https://www.mibbit.com/?url=%s
+
+# Search
+browser.search.defaultenginename=DuckDuckGo
+browser.search.order.1=DuckDuckGo
--- a/netwerk/protocol/http/nsHttpHandler.cpp	2019-05-29 17:39:53.000000000 +0200
+++ b/netwerk/protocol/http/nsHttpHandler.cpp	2019-06-07 21:14:43.057174032 +0200
@@ -485,9 +485,9 @@
       PREF_CHANGE_METHOD(nsHttpHandler::PrefsChanged), gCallbackPrefs, this);
   PrefsChanged(nullptr);
 
-  mMisc.AssignLiteral("rv:" MOZILLA_UAVERSION);
+  mMisc.AssignLiteral("rv:" MOZ_APP_UA_VERSION);
 
-  mCompatFirefox.AssignLiteral("Firefox/" MOZILLA_UAVERSION);
+  mCompatFirefox.AssignLiteral("Plasmafox/" MOZ_APP_UA_VERSION);
 
   nsCOMPtr<nsIXULAppInfo> appInfo =
       do_GetService("@mozilla.org/xre/app-info;1");
@@ -939,18 +939,14 @@
   mUserAgent += '/';
   mUserAgent += mProductSub;
 
-  bool isFirefox = mAppName.EqualsLiteral("Firefox");
-  if (isFirefox || mCompatFirefoxEnabled) {
-    // "Firefox/x.y" (compatibility) app token
-    mUserAgent += ' ';
-    mUserAgent += mCompatFirefox;
-  }
-  if (!isFirefox) {
-    // App portion
-    mUserAgent += ' ';
-    mUserAgent += mAppName;
-    mUserAgent += '/';
-    mUserAgent += mAppVersion;
+  // App portion
+  mUserAgent += ' ';
+  mUserAgent += "Firefox";
+  mUserAgent += '/';
+  mUserAgent += mAppVersion;
+  if (mCompatFirefoxEnabled) {
+       mUserAgent += ' ';
+		mUserAgent += mCompatFirefox;
   }
 }
 
--- a/browser/extensions/moz.build	2019-05-29 17:38:45.000000000 +0200
+++ b/browser/extensions/moz.build	2019-06-08 08:01:48.271294119 +0200
@@ -8,7 +8,6 @@
     'formautofill',
     'fxmonitor',
     'pdfjs',
-    'screenshots',
     'webcompat',
     'webcompat-reporter'
 ]
--- a/browser/moz.configure	2019-05-29 17:38:45.000000000 +0200
+++ b/browser/moz.configure	2019-06-07 21:35:18.363555713 +0200
@@ -5,7 +5,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 imply_option('MOZ_PLACES', True)
-imply_option('MOZ_SERVICES_HEALTHREPORT', True)
+imply_option('MOZ_SERVICES_HEALTHREPORT', False)
 imply_option('MOZ_SERVICES_SYNC', True)
 imply_option('MOZ_DEDICATED_PROFILES', True)
 imply_option('MOZ_BLOCK_PROFILE_DOWNGRADE', True)
 
--- a/modules/libpref/init/all.js	2019-05-29 17:39:35.000000000 +0200
+++ b/modules/libpref/init/all.js	2019-06-08 14:27:59.109555227 +0200
@@ -16,6 +16,7 @@
 
 pref("keyword.enabled", false);
 pref("general.useragent.compatMode.firefox", false);
+pref("general.useragent.override.addons.mozilla.org", "Mozilla/5.0 (X11; Linux x86_64; rv:60.0) Gecko/20100101 Firefox/60.0");
 
 // This pref exists only for testing purposes. In order to disable all
 // overrides by default, don't initialize UserAgentOverrides.jsm.
@@ -598,10 +599,10 @@
 
 // Whether to autostart a media element with an |autoplay| attribute.
 // ALLOWED=0, BLOCKED=1, defined in dom/media/Autoplay.idl
-pref("media.autoplay.default", 0);
+pref("media.autoplay.default", 1);
 
 // By default, don't block WebAudio from playing automatically.
-pref("media.autoplay.block-webaudio", false);
+pref("media.autoplay.block-webaudio", true);
 
 // By default, don't block muted media from playing automatically.
 pref("media.autoplay.allow-muted", true);
@@ -1453,6 +1454,8 @@
 pref("privacy.resistFingerprinting.reduceTimerPrecision.microseconds", 1000);
 // Enable jittering the clock one precision value forward
 pref("privacy.resistFingerprinting.reduceTimerPrecision.jitter", true);
+// enable resist by default
+pref("privacy.resistFingerprinting", true);
 
 pref("dom.event.contextmenu.enabled",       true);
 pref("dom.event.coalesce_mouse_move",       true);
@@ -5759,7 +5762,7 @@
 // The URL of the Firefox Accounts auth server backend
 pref("identity.fxaccounts.auth.uri", "https://api.accounts.firefox.com/v1");
 
-pref("beacon.enabled", true);
+pref("beacon.enabled", false);
 
 // UDPSocket API
 pref("dom.udpsocket.enabled", false);
--- a/browser/app/profile/firefox.js	2019-07-06 03:48:29.000000000 +0200
+++ b/browser/app/profile/firefox.js	2019-07-15 20:43:22.600524097 +0200
@@ -159,7 +159,7 @@
 
 // If set to true, the Update Service will apply updates in the background
 // when it finishes downloading them.
-pref("app.update.staging.enabled", true);
+pref("app.update.staging.enabled", false);
 
 // Update service URL:
 pref("app.update.url", "https://aus5.mozilla.org/update/6/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%SYSTEM_CAPABILITIES%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/update.xml");
@@ -204,7 +204,7 @@
 #endif
 
 // UI tour experience.
-pref("browser.uitour.enabled", true);
+pref("browser.uitour.enabled", false);
 pref("browser.uitour.loglevel", "Error");
 pref("browser.uitour.requireSecure", true);
 pref("browser.uitour.themeOrigin", "https://addons.mozilla.org/%LOCALE%/firefox/themes/");
@@ -324,7 +324,6 @@
 pref("browser.urlbar.maxCharsForSearchSuggestions", 20);
 
 pref("browser.urlbar.formatting.enabled", true);
-pref("browser.urlbar.trimURLs", true);
 
 pref("browser.urlbar.oneOffSearches", true);
 
@@ -382,7 +382,7 @@
 pref("browser.search.searchEnginesURL",      "https://addons.mozilla.org/%LOCALE%/firefox/search-engines/");
 
 // Market-specific search defaults
-pref("browser.search.geoSpecificDefaults", true);
+pref("browser.search.geoSpecificDefaults", false);
 pref("browser.search.geoSpecificDefaults.url", "https://search.services.mozilla.com/1/%APP%/%VERSION%/%CHANNEL%/%LOCALE%/%REGION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%");
 
 // search bar results always open in a new tab
@@ -480,9 +480,7 @@
 pref("browser.tabs.tabMinWidth", 76);
 // Initial titlebar state is managed by -moz-gtk-csd-hide-titlebar-by-default
 // on Linux.
-#ifndef UNIX_BUT_NOT_MAC
 pref("browser.tabs.drawInTitlebar", true);
-#endif
 
 // Offer additional drag space to the user. The drag space
 // will only be shown if browser.tabs.drawInTitlebar is true.
@@ -771,11 +769,7 @@
 // 0 goes Back/Forward
 // 1 act like PgUp/PgDown
 // 2 and other values, nothing
-#ifdef UNIX_BUT_NOT_MAC
-pref("browser.backspace_action", 2);
-#else
 pref("browser.backspace_action", 0);
-#endif
 
 // this will automatically enable inline spellchecking (if it is available) for
 // editable elements in HTML
@@ -976,17 +970,17 @@
      "https://www.mozilla.org/legal/privacy/firefox.html#crash-reporter");
 
 // base URL for web-based support pages
-pref("app.support.baseURL", "https://support.mozilla.org/1/firefox/%VERSION%/%OS%/%LOCALE%/");
+pref("app.support.baseURL", "https://github.com/torvic9/plasmafox");
 
 // base url for web-based feedback pages
 #ifdef MOZ_DEV_EDITION
-pref("app.feedback.baseURL", "https://input.mozilla.org/%LOCALE%/feedback/firefoxdev/%VERSION%/");
+pref("app.feedback.baseURL", "https://github.com/torvic9/plasmafox");
 #else
-pref("app.feedback.baseURL", "https://input.mozilla.org/%LOCALE%/feedback/%APP%/%VERSION%/");
+pref("app.feedback.baseURL", "https://github.com/torvic9/plasmafox");
 #endif
 
 // base URL for web-based marketing pages
-pref("app.productInfo.baseURL", "https://www.mozilla.org/firefox/features/");
+pref("app.productInfo.baseURL", "https://github.com/torvic9/plasmafox");
 
 // Name of alternate about: page for certificate errors (when undefined, defaults to about:neterror)
 pref("security.alternate_certificate_error_page", "certerror");
@@ -1394,10 +1388,10 @@
 // All the Geolocation preferences are here.
 //
 #ifndef EARLY_BETA_OR_EARLIER
-pref("geo.wifi.uri", "https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_LOCATION_SERVICE_API_KEY%");
+pref("geo.wifi.uri", "");
 #else
 // Use MLS on Nightly and early Beta.
-pref("geo.wifi.uri", "https://location.services.mozilla.com/v1/geolocate?key=%MOZILLA_API_KEY%");
+pref("geo.wifi.uri", "");
 #endif
 
 #ifdef XP_MACOSX
@@ -1421,7 +1421,7 @@
 
 // If set to false, FxAccounts and Sync will be unavailable.
 // A restart is mandatory after flipping that preference.
-pref("identity.fxaccounts.enabled", true);
+pref("identity.fxaccounts.enabled", false);
 
 // The remote FxA root content URL. Must use HTTPS.
 pref("identity.fxaccounts.remote.root", "https://accounts.firefox.com/");
@@ -1452,8 +1446,8 @@
 
 // URLs for promo links to mobile browsers. Note that consumers are expected to
 // append a value for utm_campaign.
-pref("identity.mobilepromo.android", "https://www.mozilla.org/firefox/android/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_campaign=");
-pref("identity.mobilepromo.ios", "https://www.mozilla.org/firefox/ios/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_campaign=");
+pref("identity.mobilepromo.android", "");
+pref("identity.mobilepromo.ios", "");
 
 // Migrate any existing Firefox Account data from the default profile to the
 // Developer Edition profile.
@@ -1464,7 +1464,7 @@
 #endif
 
 // If activated, send tab will use the new FxA commands backend.
-pref("identity.fxaccounts.commands.enabled", true);
+pref("identity.fxaccounts.commands.enabled", false);
 // How often should we try to fetch missed FxA commands on sync (in seconds).
 // Default is 24 hours.
 pref("identity.fxaccounts.commands.missed.fetch_interval", 86400);
@@ -1507,12 +1501,8 @@
 // Set Firefox to block autoplay, asking for permission by default.
 pref("media.autoplay.default", 1); // 0=Allowed, 1=Blocked
 
-#ifdef NIGHTLY_BUILD
 // Block WebAudio from playing automatically.
 pref("media.autoplay.block-webaudio", true);
-#else
-pref("media.autoplay.block-webaudio", false);
-#endif
 
 // Play with different values of the decay time and get telemetry,
 // 0 means to randomize (and persist) the experiment value in users' profiles,
@@ -1528,30 +1518,26 @@
 
 // Telemetry settings.
 // Determines if Telemetry pings can be archived locally.
-pref("toolkit.telemetry.archive.enabled", true);
+pref("toolkit.telemetry.archive.enabled", false);
 // Enables sending the shutdown ping when Firefox shuts down.
-pref("toolkit.telemetry.shutdownPingSender.enabled", true);
+pref("toolkit.telemetry.shutdownPingSender.enabled", false);
 // Enables sending the shutdown ping using the pingsender from the first session.
 pref("toolkit.telemetry.shutdownPingSender.enabledFirstSession", false);
 // Enables sending a duplicate of the first shutdown ping from the first session.
-pref("toolkit.telemetry.firstShutdownPing.enabled", true);
+pref("toolkit.telemetry.firstShutdownPing.enabled", false);
 // Enables sending the 'new-profile' ping on new profiles.
-pref("toolkit.telemetry.newProfilePing.enabled", true);
+pref("toolkit.telemetry.newProfilePing.enabled", false);
 // Enables sending 'update' pings on Firefox updates.
-pref("toolkit.telemetry.updatePing.enabled", true);
+pref("toolkit.telemetry.updatePing.enabled", false);
 // Enables sending 'bhr' pings when the browser hangs.
-pref("toolkit.telemetry.bhrPing.enabled", true);
+pref("toolkit.telemetry.bhrPing.enabled", false);
 // Enables using Hybrid Content Telemetry from Mozilla privileged pages.
-pref("toolkit.telemetry.hybridContent.enabled", true);
+pref("toolkit.telemetry.hybridContent.enabled", false);
 // Whether to enable Ecosystem Telemetry, requires a restart.
-#ifdef NIGHTLY_BUILD
-pref("toolkit.telemetry.ecosystemtelemetry.enabled", true);
-#else
 pref("toolkit.telemetry.ecosystemtelemetry.enabled", false);
-#endif
 
 // Ping Centre Telemetry settings.
-pref("browser.ping-centre.telemetry", true);
+pref("browser.ping-centre.telemetry", false);
 pref("browser.ping-centre.log", false);
 pref("browser.ping-centre.staging.endpoint", "https://onyx_tiles.stage.mozaws.net/v3/links/ping-centre");
 pref("browser.ping-centre.production.endpoint", "https://tiles.services.mozilla.com/v3/links/ping-centre");
@@ -1635,19 +1621,12 @@
 // Always enable newtab segregation using containers
 pref("privacy.usercontext.about_newtab_segregation.enabled", true);
 // Enable Contextual Identity Containers
-#ifdef NIGHTLY_BUILD
 pref("privacy.userContext.enabled", true);
 pref("privacy.userContext.ui.enabled", true);
 
 // 0 disables long press, 1 when clicked, the menu is shown, 2 the menu is shown after X milliseconds.
 pref("privacy.userContext.longPressBehavior", 2);
-#else
-pref("privacy.userContext.enabled", false);
-pref("privacy.userContext.ui.enabled", false);
 
-// 0 disables long press, 1 when clicked, the menu is shown, 2 the menu is shown after X milliseconds.
-pref("privacy.userContext.longPressBehavior", 0);
-#endif
 pref("privacy.userContext.extension", "");
 
 // Start the browser in e10s mode
@@ -1666,7 +1645,7 @@
 pref("browser.tabs.remote.warmup.unloadDelayMs", 2000);
 
 // For the about:tabcrashed page
-pref("browser.tabs.crashReporting.sendReport", true);
+pref("browser.tabs.crashReporting.sendReport", false);
 pref("browser.tabs.crashReporting.includeURL", false);
 pref("browser.tabs.crashReporting.requestEmail", false);
 pref("browser.tabs.crashReporting.emailMe", false);
@@ -1735,8 +1714,8 @@
 pref("dom.mozBrowserFramesEnabled", true);
 
 pref("extensions.pocket.api", "api.getpocket.com");
-pref("extensions.pocket.enabled", true);
-pref("extensions.pocket.oAuthConsumerKey", "40249-e88c401e1b1f2242d9e441c4");
+pref("extensions.pocket.enabled", false);
+pref("extensions.pocket.oAuthConsumerKey", "");
 pref("extensions.pocket.site", "getpocket.com");
 
 pref("signon.schemeUpgrades", true);
@@ -1822,16 +1821,12 @@
 // Normandy client preferences
 pref("app.normandy.api_url", "https://normandy.cdn.mozilla.net/api/v1");
 pref("app.normandy.dev_mode", false);
-pref("app.normandy.enabled", true);
+pref("app.normandy.enabled", false);
 pref("app.normandy.first_run", true);
 pref("app.normandy.logging.level", 50); // Warn
 pref("app.normandy.run_interval_seconds", 21600); // 6 hours
 pref("app.normandy.shieldLearnMoreUrl", "https://support.mozilla.org/1/firefox/%VERSION%/%OS%/%LOCALE%/shield");
-#ifdef MOZ_DATA_REPORTING
-pref("app.shield.optoutstudies.enabled", true);
-#else
 pref("app.shield.optoutstudies.enabled", false);
-#endif
 
 // Multi-lingual preferences
 #ifdef RELEASE_OR_BETA
@@ -1889,3 +1868,18 @@
 #else
 pref("corroborator.enabled", true);
 #endif
+
+// Overrides
+pref("devtools.onboarding.telemetry.logged", false);
+pref("toolkit.telemetry.unified", false);
+pref("toolkit.telemetry.enabled", false);
+pref("toolkit.telemetry.server", "");
+pref("datareporting.healthreport.infoURL", "");
+pref("datareporting.healthreport.uploadEnabled", false);
+pref("media.peerconnection.enabled", false);
+pref("media.peerconnection.ice.default_address_only", true);
+pref("media.peerconnection.ice.no_host", true);
+pref("browser.urlbar.trimURLs", false);
+pref("browser.newtabpage.enhanced", false);
+pref("browser.newtabpage.directory.source", "data:application/json,{}");
+pref("browser.tabs.drawInTitlebar", true);
