From 677c348f921a505def91380c1be802d6abededf7 Mon Sep 17 00:00:00 2001
From: torvic9 <torvic9@mailbox.org>
Date: Sat, 29 Aug 2020 23:38:18 +0200
Subject: [PATCH] change user agent handling

Signed-off-by: torvic9 <torvic9@mailbox.org>
---
 netwerk/protocol/http/nsHttpHandler.cpp | 24 ++++++++++--------------
 1 file changed, 10 insertions(+), 14 deletions(-)

diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 87b4461..ba881e8 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -485,11 +485,11 @@ nsresult nsHttpHandler::Init() {
   PrefsChanged(nullptr);
   Telemetry::ScalarSet(
       Telemetry::ScalarID::NETWORKING_HTTP3_ENABLED, mHttp3Enabled);
 
-  mMisc.AssignLiteral("rv:" MOZILLA_UAVERSION);
+  mMisc.AssignLiteral("rv:" MOZ_APP_UA_VERSION);
 
-  mCompatFirefox.AssignLiteral("Firefox/" MOZILLA_UAVERSION);
+  mCompatFirefox.AssignLiteral("Plasmafox/" MOZ_APP_UA_VERSION);
 
   nsCOMPtr<nsIXULAppInfo> appInfo =
       do_GetService("@mozilla.org/xre/app-info;1");
 
@@ -942,20 +942,16 @@ void nsHttpHandler::BuildUserAgent() {
   mUserAgent += mProduct;
   mUserAgent += '/';
   mUserAgent += mProductSub;
 
-  bool isFirefox = mAppName.EqualsLiteral("Firefox");
-  if (isFirefox || mCompatFirefoxEnabled) {
-    // "Firefox/x.y" (compatibility) app token
-    mUserAgent += ' ';
-    mUserAgent += mCompatFirefox;
-  }
-  if (!isFirefox) {
-    // App portion
-    mUserAgent += ' ';
-    mUserAgent += mAppName;
-    mUserAgent += '/';
-    mUserAgent += mAppVersion;
+  // App portion
+  mUserAgent += ' ';
+  mUserAgent += "Firefox";
+  mUserAgent += '/';
+  mUserAgent += mAppVersion;
+  if (mCompatFirefoxEnabled) {
+		mUserAgent += ' ';
+		mUserAgent += mCompatFirefox;
   }
 }
 
 #ifdef XP_WIN
-- 
2.28.0

