From 33ecf784876064c9c5543f7788d020bf7201897e Mon Sep 17 00:00:00 2001
From: torvic9 <torvic9@mailbox.org>
Date: Mon, 16 Nov 2020 17:51:58 +0100
Subject: [PATCH] allow webgpu on release

Signed-off-by: torvic9 <torvic9@mailbox.org>
---
 gfx/thebes/gfxPlatform.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/gfx/thebes/gfxPlatform.cpp b/gfx/thebes/gfxPlatform.cpp
index cab3568..92ae17e 100644
--- a/gfx/thebes/gfxPlatform.cpp
+++ b/gfx/thebes/gfxPlatform.cpp
@@ -2865,13 +2865,8 @@ void gfxPlatform::InitWebRenderConfig() {
 
 void gfxPlatform::InitWebGPUConfig() {
   FeatureState& feature = gfxConfig::GetFeature(Feature::WEBGPU);
   feature.SetDefaultFromPref("dom.webgpu.enabled", true, false);
-#ifndef NIGHTLY_BUILD
-  feature.ForceDisable(FeatureStatus::Blocked,
-                       "WebGPU can only be enabled in nightly",
-                       "WEBGPU_DISABLE_NON_NIGHTLY"_ns);
-#endif
 }
 
 void gfxPlatform::InitOMTPConfig() {
   ScopedGfxFeatureReporter reporter("OMTP");
